src/core/Akka/Actor/TypedActor.cs

```diff
@@ -34,7 +35,10 @@ public abstract class TypedActor : ActorBase
         /// <returns>TBD</returns>
         protected sealed override bool Receive(object message)
         {
-            MethodInfo method = GetType().GetMethod("Handle", new[] {message.GetType()});
+            Type[] msgTypeArray = new[] {message.GetType()};
+            Type actorType = GetType();
+            MethodInfo method = actorType.GetMethod("Handle", msgTypeArray) ??
+                                actorType.GetMethod($"Akka.Actor.IHandle<{msgTypeArray[0].FullName}>.Handle", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
1. This could potentially fail,  if message type is not inside the same assembly. In some cases fully qualified type name is not enough, you'll need assembly name as well.
2. This is a reflection call on each message received. Aside the fact that we (hopefully) will remove `TypedActor` in near future, its usage is heavily discouraged for this exact reason - reflection makes it order of magnitude slower than using `ReceiveActor`.
3. You could potentially patch it by constructing a dictionary of `Dictionary<Type, MethodInfo>` i.e. inside actor constructor. This way you could use reflection only once when actor is created or even try to move this to an actor static constructor.